### ISSUE TITLE
May be Epoch(val)?

### ISSUE THREAD

**Issue by hust-nj** [open-mmlab/mmcv#253]
Hi, when I use mmcv to train imagenet classification, it will print something like this(taken cifar10 example)
```text
2020-04-28 17:05:41,729 - INFO - Epoch [1][700/782]     lr: 0.10000, eta: 0:03:45, time: 0.049, data_time: 0.002, memory: 402, loss: 1.6594, acc_top1: 38.3125, acc_top5: 88.9062
2020-04-28 17:05:44,121 - INFO - Epoch [1][750/782]     lr: 0.10000, eta: 0:03:40, time: 0.048, data_time: 0.002, memory: 402, loss: 1.6392, acc_top1: 39.8750, acc_top5: 88.9062
2020-04-28 17:05:50,121 - INFO - Epoch(train) [1][157]  loss: 1.6123, acc_top1: 40.4100, acc_top5: 89.8100
2020-04-28 17:05:53,214 - INFO - Epoch [2][50/782]      lr: 0.10000, eta: 0:03:28, time: 0.062, data_time: 0.010, memory: 402, loss: 1.6114, acc_top1: 39.0312, acc_top5: 90.2812
2020-04-28 17:05:55,918 - INFO - Epoch [2][100/782]     lr: 0.10000, eta: 0:03:25, time: 0.054, data_time: 0.002, memory: 402, loss: 1.5901, acc_top1: 41.875 
```
I feel strange that in validation step it prints `Epoch(train)`, I thought it should print `Epoch(val)`.
Then I looked into code in logger hooks.
https://github.com/open-mmlab/mmcv/blob/f7de63fdd0b48a56eadb717cd9260260705f0632/mmcv/runner/hooks/logger/text.py#L92-L101

I insert the code `print(runner.mode)` and `print(mode)` in it, find that `print(mode)` always show train, even when `print(runner.mode)` show val.
Is there something wrong or a bug existing?

**Comment by ecm200** [comment]
I have come across the exact same issue when using MMDetection. 

I used the training example script, setting the validation argument to false, and changing the cfg.workflow = [('train',1), ('val', 1)]

My understanding from MMCV runner.run() method is that the workflow is formatted to run these processes in order, for the number of epochs specified as int. So my interpretation of how I have set this up, is to that I will perform validation at the end of each training epoch.

This piece of code ensures that the validation dataset is loaded when a validation step is requested.
```python
datasets = [build_dataset(cfg.data.train)]
if len(cfg.workflow) == 2:
    val_dataset = copy.deepcopy(cfg.data.val)
    val_dataset.pipeline = cfg.data.train.pipeline
    datasets.append(build_dataset(val_dataset))
```
However, when the model is training, the logging returns the following, showing a training label for the second process which I interpret as the validation step, given that it is not reporting a learning rate:

```
2020-05-13 10:00:29,583 - mmdet - INFO - Start running, host: edmorris@willow-tree-cnn-gpu-lin64, work_dir: /home/edmorris/notebooks/Projects/WillowTree/Repo/advanced_seg/MMDetection_experiments/output/spherical_test_data_v1_100_100_MaskRCNN_ResNet50_FPN_13052020_095911
2020-05-13 10:00:29,583 - mmdet - INFO - workflow: [('train', 1), ('val', 1)], max: 12 epochs
2020-05-13 10:01:01,888 - mmdet - INFO - Epoch [1][50/50]       lr: 0.00025, eta: 0:05:51, time: 0.639, data_time: 0.027, memory: 4089, loss_rpn_cls: 0.4464, loss_rpn_bbox: 0.0000, loss_cls: 1.6294, acc: 77.0000, loss_bbox: 0.0000, loss: 2.0759
2020-05-13 10:01:18,338 - mmdet - INFO - Epoch(train) [1][50]   loss_rpn_cls: 0.0076, loss_rpn_bbox: 0.0000, loss_cls: 0.8283, acc: 100.0000, loss_bbox: 0.0000, loss: 0.8359
2020-05-13 10:01:50,291 - mmdet - INFO - Epoch [2][50/50]       lr: 0.00050, eta: 0:05:18, time: 0.636, data_time: 0.026, memory: 4089, loss_rpn_cls: 0.0018, loss_rpn_bbox: 0.0000, loss_cls: 1.1091, acc: 100.0000, loss_bbox: 0.0000, loss: 1.1110
2020-05-13 10:02:06,634 - mmdet - INFO - Epoch(train) [2][50]   loss_rpn_cls: 0.0004, loss_rpn_bbox: 0.0000, loss_cls: 1.2133, acc: 100.0000, loss_bbox: 0.0000, loss: 1.2138
2020-05-13 10:02:38,528 - mmdet - INFO - Epoch [3][50/50]       lr: 0.00075, eta: 0:04:46, time: 0.634, data_time: 0.027, memory: 4089, loss_rpn_cls: 0.0005, loss_rpn_bbox: 0.0000, loss_cls: 1.1461, acc: 100.0000, loss_bbox: 0.0000, loss: 1.1465
```
